### ISSUE TITLE
When casperjs tests fail with slimerjs, exit code is 0

### ISSUE THREAD

**Issue by stiemannkj1** [n1k0/casperjs#1357]
Steps to reproduce:

1. Create the following **`test.js`** file:

```
casper.test.begin("Bla", 1, function(test) {
   test.assert(false);
   test.done();
});
```
2. Run the test with `slimerjs`:
```
casperjs test --engine=slimerjs test.js
```
3. Output the exit code:
```
echo $?
```

Expected result:
The exit code should be `1` signifying a test failure.

Actual result:
The exit code is `0` signifying success.

If you run the test with `phantomjs`, it will exit with a `1`:
```
$ casperjs test test.js
...
$ echo $?
1
```

Tested on `casperjs` `1.1.0-beta3`.

**Note:** @rbelouin originally wrote these steps [in his comment](https://github.com/n1k0/casperjs/issues/314#issuecomment-146558159) on #314. I'm just rewriting his comment as its own issue since that issue was never reopened.

**Comment by istr** [comment]
Yes. It seems that [slimerjs support and testing is still incomplete](https://github.com/n1k0/casperjs/pull/1351). I will try to address these issues in a separate feature branch / PR once the support for 2.0.0 is merged into upstream.

**Comment by istr** [comment]
Duplicate of https://github.com/n1k0/casperjs/issues/904.
